### DISCUSSION TITLE
A question about tree view checkbox

### DISCUSSION THREAD

**Discussion by dddssw** [microsoft/vscode-discussions#2042] [Extension Development QnA] [answered]
![image](https://github.com/user-attachments/assets/39242691-4c09-46a0-aad5-a0ca49f631c5)
I want to click this UI command and get the checked children. How should I implement it?

**Comment by isidorn** [comment]
Well did you try Well you keep the state of each tree child, so your extension should be aware if it is checked or not.
What do you visually want to do exactly?

**Reply by dddssw** [reply, marked as answer]
Thank you for your answer. I just want to get the selected status of the sub items. My idea now is to save these states during initialization and then listen for onDudChangeCheckboxState to update the status. When using it, I will know which ones are checked. Is there a simpler method? I haven't found any other related APIs

**Reply by isidorn** [reply]
I think this is the way to go. I am not aware of a simpler method (though I do not know all the Tree APIs).

**Reply by dddssw** [reply]
All right, Thank.
